internal/server/routes.go

```diff
@@ -282,12 +283,19 @@ func (a *API) notFoundHandler(c *gin.Context) {
 	}
 
 	c.Status(http.StatusNotFound)
-	buf, err := assetFS.ReadFile("ui/404.html")
-	if err != nil {
-		logging.S.Error(err)
+
+	buf := []byte("404 not found")
+
+	const filePath404 = "ui/static/404.html"
+	uiFS := a.server.options.UI.FS
+	if _, err := fs.Stat(uiFS, filePath404); err == nil {
+		buf, err = fs.ReadFile(uiFS, filePath404)
```

Comment:
I changed this if from `!= nil`, to stat the file. If the UI static files exist then it'll use the page from the static assets. Otherwise it will use the string above.